controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -272,6 +273,16 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 		return r.scaleDownControlPlane(ctx, cluster, kcp, ownedMachines)
 	}
 
+	// Update kube-proxy image name
```

Comment:
I want to say that each of the operations above (upgrade, initialize, scale up, scale down) should already end there reconciliation with a `ctrl.Result{Requeue: true}, nil`, but it might be good to verify that, otherwise there might be an edge case where this this call could be skipped.

Reply:
Tested those scenarios, requeuing successfully when there is no error.

Reply:
great, thanks for verifying